src/ChannelServer/Network/ChannelPacketHandler.cs

```diff
@@ -806,48 +807,81 @@ public void CZ_CLIENT_HIT_LIST(ChannelConnection conn, Packet packet)
 		public void CZ_REQ_NORMAL_TX_NUMARG(ChannelConnection conn, Packet packet)
 		{
 			var size = packet.GetShort();
-			var type = packet.GetShort(); // 1 - statpints, 2 - skillpoints
-			var count = packet.GetInt();
-			//Throws exception when trying distributing skill points (count 6)
-			if (count != 5)
-				throw new Exception("Unknown CZ_REQ_NORMAL_TX_NUMARG format, expected 5 stats.");
-
+			var txType = packet.GetShort();
 			var character = conn.SelectedCharacter;
 
-			for (int i = 0; i < count; ++i)
+			switch (txType)
 			{
-				var stat = packet.GetInt();
-				if (stat == 0)
-					continue;
-
-				if (character.StatPoints < stat)
-				{
-					Log.Warning("User '{0}' tried to spent more stat points than he has.", conn.Account.Name);
+				case 1:
+					var count = packet.GetInt();
+					if (count != 5)
+						throw new Exception("Unknown CZ_REQ_NORMAL_TX_NUMARG format, expected 5 stats.");
+
+					for (int i = 0; i < count; ++i)
+					{
+						var stat = packet.GetInt();
+
+						Log.Warning("stat{1}: {0}", stat, i);
```

Comment:
A warning? What are you warning the user about? o.o If you want to log something for debug purposes you should use Debug, which is only visible in the console if you enable it in the log.conf.